monitoring/scd_qualifier/utils.py

```diff
@@ -0,0 +1,117 @@
+    return str_value
+
+class LatLngPoint(ImplicitDict):
+    '''A class to hold information about a location as Latitude / Longitude pair'''
+    lat: float
+    lng: float
+
+class Radius(ImplicitDict):
+    ''' A class to hold the radius of a circle for the outline_circle object as specified per the ASTM standard '''
+    value: float
+    units:str
+
+class VolumePolygon(ImplicitDict):
+    ''' A class to hold the polygon object, used in the outline_polygon of the Volume3D object as specified in the ASTM standard '''
+    vertices: List[LatLngPoint] # A minimum of three LatLngPoints
+
+class Circle(ImplicitDict):
+    ''' Hold the details of a circle object used in the outline_circle object as specified in the ASTM standard '''
+    center: LatLngPoint 
+    radius: Radius
+
+
+class Altitude(ImplicitDict):
+    ''' A class to hold altitude per ASTM standard '''
+    value:int
+    reference:str 
+    units: str 
+
+class Time(ImplicitDict):
+    ''' A class to hold Time per ASTM standard'''
+    value: str
+    format:str 
+
+class OperationalIntentReference(ImplicitDict):
+    """Class for keeping track of an operational intent reference"""
+    id: str
+
+class Volume3D(ImplicitDict):
+    '''A class to hold Volume3D objects '''
+    
+    outline_polygon: VolumePolygon
+    altitude_lower: Altitude
+    altitude_upper: Altitude
+
+class Volume4D(ImplicitDict):
+    '''A class to hold ASTM Volume4D objects '''
+    volume: Volume3D
+    time_start: StringBasedDateTime
+    time_end: StringBasedDateTime
+
+
+class OperationalIntentDetails(ImplicitDict):
+    """Class for holding details of an operational intent reference """
+    volumes: List[Volume4D]
+    priority: int
+
+class VolumeGenerationRule(ImplicitDict):
+    """ A class to hold configuration for holding rule for volume generation """
+    intersect_altitude: bool = 0
+    intersect_time: bool = 0
+    is_control: bool = 0
+    expected_result: str
+
+class GeoJSONFeature(ImplicitDict):
+    ''' A class to hold GeoJSON Feature flight plan, at this moment we are using a LineString but it could be a LineString or Polygon '''
+    geojson: Feature
+    
+class SCDVolume4D(ImplicitDict):
+    ''' A class to hold a volume 4D with path and time details and GeoJSON flight plan, USSP will have to transform flight plan to a  '''
+    flight_plan: GeoJSONFeature
+    time_start: StringBasedDateTime
+    time_end: StringBasedDateTime
+    altitude_lower: Altitude
+    altitude_upper: Altitude
+    
+class GeometryGenerationRule(ImplicitDict):
+    """ A class to hold configuration for developing treatment flight path """
+    intersect_space:bool = 0
+    
+class GeometryPayload(ImplicitDict):
+    ''' An object to hold generated flight path, is_control is a nomenclature used to see if the generated path is the first one '''
+    geometry: Union[LineString, Polygon]
+    is_control: bool
+    geometry_generation_rule: GeometryGenerationRule
+  
+class FlightAuthPayload(ImplicitDict):
```

Comment:
Class names should usually be a succinct form of self-documentation, and that succinct form should match the long form.  In this case, we're calling the class a flight auth payload, but then we document it as flight authorization details.  I think the long form is probably more accurate, in which case the better class name would be FlightAuthDetails.  But, I think we can do even better than that -- these are actually the details of a flight authorization, these are the details of a request for flight authorization (depending on these details, the flight authorization may never be created).  So, I would recommend naming this class FlightAuthRequestDetails, or possibly just FlightAuthRequest.

Reply:
Thank you for this, I really struggled with the naming. The only thing here is that this is not a complete request, the thing that is missing here is the Volume4D, this class holds the "data-format" checks so for the moment, I have named it `FlightAuthPartialRequestDetails`, the idea is that we add the volume 4D and then send this for USS processing. See also this [other comment](https://github.com/interuss/dss/pull/604#discussion_r753081200).